### ISSUE TITLE
rainbow atari example: load the saved policy cannot get the same Mean reward 

### ISSUE THREAD

**Issue by Moirai7** [aai-institute/tianshou#880]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gymnasium as gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```

I used the atari_rainbow.py to save the policy. 

def save_best_fn(policy):
        torch.save(policy.state_dict(), os.path.join(log_path, "policy.pth"))

After training and calling the watch(), it shows "Mean reward (over 10 episodes): 472.4"
But if I resume the policy: policy.load_state_dict(torch.load(args.resume_path, map_location=args.device))
The mean reward is 1.4. I have checked the code that the policy has been loaded(most weights such as model.net.0.weight are loaded. However, one weight "support" is not updated). I wonder why and how to load the saved policy?

**Comment by Trinkle23897** [comment]
have you checked epsilon setting?

**Comment by Moirai7** [comment]
I found out I need to create a new test collector:
test_collector = Collector(policy, test_envs, exploration_noise=True)

If I create the new test collector, the reward because 344.
In the watch function, it reset the test_collector using test_collector.reset().
It seems this function did not correct reset the collector.
